src/formatDate.js

```diff
@@ -7,8 +7,62 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparatorFormat = fromFormat[3];
+  const toSeparatorFormat = toFormat[3];
+  const dateObject = {};
+
+  const arrayFromDate = date.split(fromSeparatorFormat);
+
+  fromFormat.forEach((part, i) => {
+    switch (part) {
+      case 'YYYY':
+        dateObject.year = arrayFromDate[i];
+        break;
+
+      case 'YY':
+        const year = parseInt(arrayFromDate[i]);
+
+        dateObject.year =
+          year < 30 ? '20' + arrayFromDate[i] : '19' + arrayFromDate[i];
```

Comment:
The `parseInt` function should have a radix parameter to ensure the string is always parsed as a base-10 number. For example: `parseInt(arrayFromDate[i], 10)`.